### PR TITLE
[Fix](multi catalog statistics)Improve external table statistics collection.

### PR DESCRIPTION
**Pull request by Jibing-Li**
Improve external table statistics collection, including log, observability and fix some bugs.
1. Add Running state for statistics job.
2. Add progress for show analyze job. (n/m tasks finished, n/m task failed and so on)
3. Add analyze time cost for show analyze task.
4. Make task failure message more clear.
5. Synchronize the job status updating code in updateTaskStatus.
6. Fix NPE in HMSAnalyzeTask. (Avoid refreshing statistics cache if the collection sql failed)
7. Return error message for with sync collection while timeout. 
8. Log level improvement
9. Fix misuse of logCreateAnalysisJob for tasks. 

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by Jibing-Li** [comment]
run buildall

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 45.08 seconds
 stream load tsv:          542 seconds loaded 74807831229 Bytes, about 131 MB/s
 stream load json:         20 seconds loaded 2358488459 Bytes, about 112 MB/s
 stream load orc:          65 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          30 seconds loaded 861443392 Bytes, about 27 MB/s
 insert into select:          30.2 seconds inserted 10000000 Rows, about 331K ops/s
 storage size: 17163790131 Bytes

**Comment by Jibing-Li** [comment]
run buildall

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 45.17 seconds
 stream load tsv:          505 seconds loaded 74807831229 Bytes, about 141 MB/s
 stream load json:         20 seconds loaded 2358488459 Bytes, about 112 MB/s
 stream load orc:          65 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          31 seconds loaded 861443392 Bytes, about 26 MB/s
 insert into select:          29.2 seconds inserted 10000000 Rows, about 342K ops/s
 storage size: 17167603534 Bytes

File: fe/fe-core/src/main/java/org/apache/doris/statistics/AnalysisManager.java [hunk 1]
```diff
@@ -578,63 +580,70 @@ private void createTaskForExternalTable(AnalysisInfo jobInfo,
...
+                if (hasFailure) {
+                    job.state = AnalysisState.FAILED;
+                    logCreateAnalysisJob(job);
+                } else {
+                    job.state = AnalysisState.FINISHED;
+                    if (job.jobType.equals(JobType.SYSTEM)) {
+                        try {
+                            updateTableStats(job);
+                        } catch (Throwable e) {
+                            LOG.warn("Failed to update Table statistics in job: {}", info.toString());
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/statistics/AnalysisManager.java:640]: print `e`

File: fe/fe-core/src/main/java/org/apache/doris/statistics/AnalysisManager.java [hunk 2]
```diff
@@ -700,6 +709,29 @@ public List<AnalysisInfo> showAnalysisJob(ShowAnalyzeStmt stmt) {
...
+                    break;
+                case FAILED:
+                    failed++;
+                    break;
+                default:
+                    inProgress++;
+                    break;
+            }
+        }
+        return String.format("%d/%d finished, %d/%d failed, %d/%d in progress",
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/statistics/AnalysisManager.java:731]: How about: `finished/failed/in progress/total` ?

File: fe/fe-core/src/main/java/org/apache/doris/statistics/AnalysisManager.java [hunk 3]
```diff
@@ -813,21 +845,26 @@ public void cancel() {
...
+                    colNames.add(task.info.colName);
+                    errorMessages.add("Cancelled");
                     continue;
                 }
                 try {
                     task.doExecute();
                     updateSyncTaskStatus(task, AnalysisState.FINISHED);
                 } catch (Throwable t) {
                     colNames.add(task.info.colName);
+                    errorMessages.add(t.getMessage());
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/statistics/AnalysisManager.java:860]: Util.getRootCauseMessage(t)

File: fe/fe-core/src/main/java/org/apache/doris/statistics/AnalysisManager.java [hunk 4]
```diff
@@ -578,63 +580,70 @@ private void createTaskForExternalTable(AnalysisInfo jobInfo,
...
-            if (taskInfo.state.equals(AnalysisState.RUNNING) || taskInfo.state.equals(AnalysisState.PENDING)) {
-                allFinished = false;
-                break;
+        // Synchronize the job state change in job level.
+        synchronized (job) {
+            job.lastExecTimeInMs = time;
+            // Set the job state to RUNNING when its first task becomes RUNNING.
+            if (info.state.equals(AnalysisState.RUNNING) && job.state.equals(AnalysisState.PENDING)) {
+                job.state = AnalysisState.RUNNING;
+                logCreateAnalysisJob(job);
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/statistics/AnalysisManager.java:616]: I think we only need to log the job or task in final state(finished or cancelled)

**Comment by Jibing-Li** [comment]
run buildall

**Review by Kikyou1997 — APPROVED** [review]

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 45.11 seconds
 stream load tsv:          502 seconds loaded 74807831229 Bytes, about 142 MB/s
 stream load json:         20 seconds loaded 2358488459 Bytes, about 112 MB/s
 stream load orc:          64 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          30 seconds loaded 861443392 Bytes, about 27 MB/s
 insert into select:          29.2 seconds inserted 10000000 Rows, about 342K ops/s
 storage size: 17167424515 Bytes

**Review by morningman — APPROVED** [review]
LGTM

**Pull request merged**